src/sidebar/components/ShareDialog/test/FileInput-test.js

```diff
@@ -0,0 +1,101 @@
+import { mount } from 'enzyme';
+import { createRef } from 'preact';
+
+import FileInput from '../FileInput';
+
+describe('FileInput', () => {
+  let fakeOnFileSelected;
+  let fileInputRef;
+
+  beforeEach(() => {
+    fakeOnFileSelected = sinon.stub();
+    fileInputRef = createRef();
+  });
+
+  afterEach(() => {
+    fileInputRef.current.click.restore();
+  });
+
+  const createInput = (disabled = undefined) => {
+    const wrapper = mount(
+      <FileInput
+        onFileSelected={fakeOnFileSelected}
+        fileInputRef={fileInputRef}
+        disabled={disabled}
+      />,
+    );
+
+    // Mock ref's current after mounting element, otherwise fileInputRef.current
+    // is still null
+    sinon.stub(fileInputRef.current, 'click');
+
+    return wrapper;
+  };
+
+  const createFile = name => new File([name], `${name}.json`);
+
+  const fillInputWithFiles = (fileInput, files) => {
+    const list = new DataTransfer();
+
+    files.forEach(file => list.items.add(file));
+    fileInput.getDOMNode().files = list.files;
```

Comment:
👍 - Somehow I missed this method of programmatically constructing a file list in the `ImportAnnotations` test and relied on monkey-patching `HTMLImportElement.files` instead.